pkg/render/egressgateway/egressgateway.go

```diff
@@ -0,0 +1,470 @@
+	return &component{
+		config: config,
+	}
+}
+
+type component struct {
+	config *Config
+}
+
+// Config contains all the config information EgressGateway needs to render component.
+type Config struct {
+	PullSecrets  []*corev1.Secret
+	Installation *operatorv1.InstallationSpec
+	OSType       rmeta.OSType
+	EgressGW     *operatorv1.EgressGateway
+
+	egwImage  string
+	VXLANVNI  int
+	VXLANPort int
+
+	Openshift bool
+	// Whether or not the cluster supports pod security policies.
+	UsePSP bool
+}
+
+func (c *component) ResolveImages(is *operatorv1.ImageSet) error {
+	reg := c.config.Installation.Registry
+	path := c.config.Installation.ImagePath
+	prefix := c.config.Installation.ImagePrefix
+
+	if c.config.OSType != c.SupportedOSType() {
+		return fmt.Errorf("Egress Gateway is supported only on %s", c.SupportedOSType())
+	}
+
+	var err error
+	c.config.egwImage, err = components.GetReference(components.ComponentEgressGateway, reg, path, prefix, is)
+	return err
+}
+
+func (c *component) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeLinux
+}
+
+func (c *component) Objects() ([]client.Object, []client.Object) {
+	objectsToCreate := []client.Object{}
+	objectsToDelete := []client.Object{}
+	objectsToCreate = append(objectsToCreate, c.egwServiceAccount())
+	objectsToCreate = append(objectsToCreate, c.egwDeployment())
+	if c.config.UsePSP {
+		objectsToCreate = append(objectsToCreate, c.egwPodSecurityPolicy())
+		objectsToCreate = append(objectsToCreate, c.egwClusterRole())
+		objectsToCreate = append(objectsToCreate, c.egwClusterRoleBinding())
+	} else if c.config.Openshift {
+		objectsToCreate = append(objectsToCreate, c.egwSecurityContextConstraints())
+	} else {
+		objectsToDelete = append(objectsToDelete, c.egwClusterRole())
+		objectsToDelete = append(objectsToDelete, c.egwClusterRoleBinding())
+	}
+	return objectsToCreate, objectsToDelete
+}
+
+func (c *component) Ready() bool {
+	return true
+}
+
+func (c *component) egwDeployment() *appsv1.Deployment {
+	d := appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{Kind: "Deployment", APIVersion: "apps/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      c.config.EgressGW.Name,
+			Namespace: c.config.EgressGW.Namespace,
+			Labels:    c.config.EgressGW.Spec.Template.Metadata.Labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: c.config.EgressGW.Spec.Replicas,
+			Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"k8s-app": c.config.EgressGW.Name}},
+			Template: *c.deploymentPodTemplate(),
+		},
+	}
+
+	if overrides := c.config.EgressGW; overrides != nil {
+		rcomp.ApplyDeploymentOverrides(&d, overrides)
+	}
+
+	return &d
+}
+
+func (c *component) deploymentPodTemplate() *corev1.PodTemplateSpec {
+	var ps []corev1.LocalObjectReference
+	for _, x := range c.config.PullSecrets {
+		ps = append(ps, corev1.LocalObjectReference{Name: x.Name})
+	}
+	return &corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: c.egwBuildAnnotations(),
```

Comment:
Do we not need to provide labels for the pods?

Reply:
Should be handled by ```GetPodTemplateMetadata```